galpy/potential_src/potential_c_ext/SpiralArmsPotential.c

```diff
@@ -0,0 +1,763 @@
+#include <math.h>
+#include <galpy_potentials.h>
+
+double gam(double R, double phi, double N, double phi_ref, double r_ref, double tan_alpha);
+
+double dgam_dR(double R, double N, double tan_alpha);
+
+double K(double R, double n, double N, double sin_alpha);
+
+double B(double R, double H, double n, double N, double sin_alpha);
+
+double D(double R, double H, double n, double N, double sin_alpha);
+
+double dK_dR(double R, double n, double N, double sin_alpha);
+
+double dB_dR(double R, double H, double n, double N, double sin_alpha);
+
+double dD_dR(double R, double H, double n, double N, double sin_alpha);
+
+double SpiralArmsPotentialEval(double R, double z, double phi, double t,
+                               struct potentialArg *potentialArgs) {
+    // Get args
+    double *args = potentialArgs->args;
+
+    int nCs = (int) *args++;
+    double amp = *args++;
+    double N = *args++;
+    double sin_alpha = *args++;
+    double tan_alpha = *args++;
+    double r_ref = *args++;
+    double phi_ref = *args++;
+    double Rs = *args++;
+    double H = *args++;
+    double omega = *args++;
+
+    phi = phi - omega * t;
+
+    double g = gam(R, phi, N, phi_ref, r_ref, tan_alpha);
+
+    // Return value of the potential.
+    double sum = 0;
+
+    int n;
+    for (n = 1; n <= nCs; n++) {
+        double Cn = *args++;
+        double Kn = K(R, n, N, sin_alpha);
+        double Bn = B(R, H, n, N, sin_alpha);
+        double Dn = D(R, H, n, N, sin_alpha);
+
+        sum += Cn / Kn / Dn * cos(n * g) / pow(cosh(Kn * z / Bn), Bn);
+    }
+
+    return -amp * H * exp(-(R - r_ref) / Rs) * sum;
+}
+
+double SpiralArmsPotentialRforce(double R, double z, double phi, double t,
+                                 struct potentialArg *potentialArgs) {
+
+    // Get args
+    double *args = potentialArgs->args;
+
+    int nCs = (int) *args++;
+    double amp = *args++;
+    double N = *args++;
+    double sin_alpha = *args++;
+    double tan_alpha = *args++;
+    double r_ref = *args++;
+    double phi_ref = *args++;
+    double Rs = *args++;
+    double H = *args++;
+    double omega = *args++;
+
+    phi = phi - omega * t;
```

Comment:
Can't we just pass ``phi-omega*t`` to the g function, rather than assigning ``phi=phi-omega*t``? I think ``phi`` is only used once.